code/frontend/src/App.js

```diff
@@ -15,7 +16,7 @@ function App() {
           </div>
           <div className="AppSignInButtonContainerWrapper">
             <Link to="/search">
-              <button data-cy="signInButton">Sign-In with Google</button>
```

Comment:
Update `onClick={signInWithGoogle}` to call a method which checks if the user is already logged in. if yes then show the search component. if not then run the `signInWithGoogle` function

To Check if the user is authenticated or not, you can do something like this 

import { getAuth } from "firebase/auth";
getAuth().onAuthStateChanged((userCred) => {
      if(userCred){
           // user is authenticated
      }
    })